tests/Unit/Domain/Speaker/SpeakerProfileTest.php

```diff
@@ -0,0 +1,627 @@
+    {
+        $hiddenProperties = [
+            'url',
+        ];
+        
+        $speaker = $this->createUserMock();
+
+        $profile = new SpeakerProfile($speaker, $hiddenProperties);
+
+        $this->expectException(NotAllowedException::class);
+
+        $profile->getUrl();
+    }
+
+    public function testGetUrlReturnsUrlIfPropertyIsNotHidden()
+    {
+        $url = $this->getFaker()->url;
+
+        $speaker = $this->createUserMock([
+            'url' => $url,
+        ]);
+
+        $profile = new SpeakerProfile($speaker);
+
+        $this->assertSame($url, $profile->getUrl());
+    }
+
+    public function testGetInfoThrowsNotAllowedExceptionIfPropertyIsHidden()
+    {
+        $hiddenProperties = [
+            'info',
+        ];
+        
+        $speaker = $this->createUserMock();
+
+        $profile = new SpeakerProfile($speaker, $hiddenProperties);
+
+        $this->expectException(NotAllowedException::class);
+
+        $profile->getInfo();
+    }
+
+    public function testGetInfoReturnsInfoIfPropertyIsNotHidden()
+    {
+        $info = $this->getFaker()->text;
+
+        $speaker = $this->createUserMock([
+            'info' => $info,
+        ]);
+
+        $profile = new SpeakerProfile($speaker);
+
+        $this->assertSame($info, $profile->getInfo());
+    }
+
+    public function testGetBioThrowsNotAllowedExceptionIfPropertyIsHidden()
+    {
+        $hiddenProperties = [
+            'bio',
+        ];
+        
+        $speaker = $this->createUserMock();
+
+        $profile = new SpeakerProfile($speaker, $hiddenProperties);
+
+        $this->expectException(NotAllowedException::class);
+
+        $profile->getBio();
+    }
+
+    public function testGetBioReturnsBioIfPropertyIsNotHidden()
+    {
+        $bio = $this->getFaker()->text;
+
+        $speaker = $this->createUserMock([
+            'bio' => $bio,
+        ]);
+
+        $profile = new SpeakerProfile($speaker);
+
+        $this->assertSame($bio, $profile->getBio());
+    }
+    
+    public function testGetTransportationThrowsNotAllowedExceptionIfPropertyIsHidden()
+    {
+        $hiddenProperties = [
+            'transportation',
+        ];
+        
+        $speaker = $this->createUserMock();
+
+        $profile = new SpeakerProfile($speaker, $hiddenProperties);
+
+        $this->expectException(NotAllowedException::class);
+
+        $profile->getTransportation();
+    }
+
+    /**
+     * @dataProvider providerDoesNotNeedTransportation
+     *
+     * @param mixed $transportation
+     */
+    public function testGetTransportationReturnsFalseIfPropertyIsNotHidden($transportation)
+    {
+        $speaker = $this->createUserMock([
+            'transportation' => $transportation,
+        ]);
+
+        $profile = new SpeakerProfile($speaker);
+
+        $this->assertFalse($profile->getTransportation());
+    }
+
+    public function providerDoesNotNeedTransportation(): \Generator
```

Comment:
I'm using a data provider here because we are using a non-strict comparison in [`SpeakerProfile::getTransportation()`](https://github.com/opencfp/opencfp/blob/40ce73ce014afec8faf92443401688ee8975d6b9/classes/Domain/Speaker/SpeakerProfile.php#L113-L118).

Ideally we would be using a strict comparison, and then the data provider would become obsolete.